urlc1c8a2efc1_wydler_mdbt_tgz-pJ8-de_hrw_mdbt_CRSJ8/src/de/hrw/mdbt/CRSOperations.java

```diff
@@ -0,0 +1,209 @@
+		PriceClass pc1 = new PriceClass("TooCheapToDrive", "Good Luck!", 10, 2, 3);
+		VehicleGroup vg1a = new VehicleGroup(100, "Diesel", c1a.getTime(), "Blue Metallic", m1, pc1);
+		VehicleGroup vg1b = new VehicleGroup(100, "Diesel", c1b.getTime(), "Green Metallic", m1, pc1);
+		VehicleGroup vg1c = new VehicleGroup(100, "Diesel", c1c.getTime(), "Red Metallic", m1, pc1);
+		Vehicle v1 = new Vehicle("RV-DE-0001","1",1,b1,vg1a);
+		new Rental(c1a.getTime(), c1b.getTime(), v1, p1);
+		new Rental(c1c.getTime(), c1d.getTime(), v1, cy1);
+		new Vehicle("RV-FA-0002","2",10,b1,vg1b);
+		new Vehicle("RV-UL-0003","3",100,b1,vg1b);
+		new Vehicle("RV-T-0004","4",1000,b1,vg1c);
+
+		Calendar c2 = new GregorianCalendar(2012, 1, 2);
+		Model m2 = new Model();
+		PriceClass pc2 = new PriceClass("TooSlowToDrive", "Take Your Time!", 1, 10, 100);
+		VehicleGroup vg2 = new VehicleGroup(10, "Benzin", c2.getTime(), "Blue Metallic", m2, pc2);
+		new Vehicle("RV-DE-0005","1",1,b1,vg2);
+		new Vehicle("RV-FA-0006","2",10,b1,vg2);
+		new Vehicle("RV-UL-0007","3",100,b1,vg2);
+		new Vehicle("RV-TY-0008","4",1000,b1,vg2);
+		db.store(b1);
+
+		Branch b2 = new Branch("AnotherDefaultBranch",
+				new Address("DefaultState", "AnotherDefaultCity", "DefaultStreet", "234567", "DefaultType"),
+				Time.valueOf("8:0:0"), Time.valueOf("16:0:0"),
+				"0234567891");
+		Calendar c3 = new GregorianCalendar(2013, 1, 2);
+		LicenseClass l2 = new LicenseClass("TestPilot","WhatIsThisButtonFor?");
+		Model m3 =  new Model("Black&Decker","NX01",200,"metric",l2);
+		PriceClass pc3 = new PriceClass("IAmBatMan", "NANANANANANANANANANANANANANANANANANA BATMAAAN!!!", 0, 999, 9999);
+		VehicleGroup vg3 = new VehicleGroup(100, "Diesel", c3.getTime(), "Blue Metallic", m3, pc3);
+		Vehicle v2 = new Vehicle("FN-BM-0001","1",1,b2,vg3);
+		new Rental(c1a.getTime(), c1d.getTime(), v2, cy1);
+		db.store(b2);
+
+		db.commit();
+	}
+
+	public static ObjectSet<VehicleGroup> createOffer(ObjectContainer db, final Branch b, final Date startDate, final Date endDate, final PriceClass pc) {
+		if(b==null || pc==null || startDate.after(endDate) )
+			return null;
+		@SuppressWarnings("serial")
+		ObjectSet<VehicleGroup> result = db.query(new Predicate<VehicleGroup>() {
+			@Override
+			public boolean match(VehicleGroup vg) {
+				if( vg.getPriceClass() != pc) return false;
+				for(Vehicle v : vg.getVehicles()) {
+					if(v.getBranch() != b) continue;
+					boolean overlap = false;
+					for(Rental r : v.getRentals()) {
+						// check next vehicle if any rental overlaps
+						if(r.getStartDate().getTime() <= endDate.getTime() && startDate.getTime() <= r.getEndDate().getTime())
+						{
+							overlap = true;
+							break;
+						}
+					}
+					if(!overlap)
+						return true;
+				}
+				// no possible vehicles found
+				return false;
+			}
+		});
+		return result;
+	}
+
+	public static boolean createReservation(ObjectContainer db, Customer c, Branch b, VehicleGroup vg, Date startDate, Date endDate) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.